cluster/rbac/manager.go

```diff
@@ -148,17 +149,42 @@ func (m *Manager) UpsertRolesPermissions(c *cmd.ApplyRequest) error {
 
 			// create new permissions
 			for idx := range policies {
-				if req.Roles[roleName][idx].Domain != authorization.SchemaDomain {
-					continue
+				if req.Roles[roleName][idx].Domain == authorization.SchemaDomain {
+					parts := strings.Split(req.Roles[roleName][idx].Resource, "/")
+					if len(parts) < 3 {
+						// shall never happens
+						return fmt.Errorf("invalid schema path")
+					}
+					req.Roles[roleName][idx].Resource = authorization.CollectionsMetadata(parts[2])[0]
 				}
 
-				parts := strings.Split(req.Roles[roleName][idx].Resource, "/")
-				if len(parts) < 3 {
-					// shall never happens
-					return fmt.Errorf("invalid schema path")
+				if req.Roles[roleName][idx].Domain == authorization.RolesDomain &&
+					req.Roles[roleName][idx].Verb == conv.CRUD {
+					// this will override any role was created before 1.28
+					// to reset default to
+					req.Roles[roleName][idx].Verb = authorization.ROLE_SCOPE_MATCH
```

Comment:
is this a copy from the new code below? Maybe we should do something like:
```
while v<LATEST:
   switch :
    case v0 :
      ..upgrade to v1...
   case v1:
      ---upgrade to v2....
```

to keep this mantainable 🤔 Also fine to do that in a follow up PR

Reply:
it's cant be multiple versions 
the idea behind migration, that RAFT command will always use latest, however with any change we add new iota and automatically latest version increases 

```
switch req.Version {
 case version x 
   do something 
   do something else from version x + 1
  case version x+1 
    do something else 
}
```

Reply:
i'd like to keep it explicit for now, to make sure nothing is missed , but i see your point we can do func to do the version needed changes